pkg/operator/ceph/controller/predicate.go

```diff
@@ -528,6 +516,32 @@ func WatchPredicateForNonCRDObject(owner runtime.Object, scheme *runtime.Scheme)
 					return false
 				}
 
+				// If the resource is an OBC we check only for changes needed for bucket notifications
+				// meaning: label changes, actual bucket name was set or state changes to "bound"
+				newObj, ok := e.ObjectNew.(*bktv1alpha1.ObjectBucketClaim)
+				if ok {
+					oldObj, _ := e.ObjectOld.(*bktv1alpha1.ObjectBucketClaim)
+					diff := cmp.Diff(
+						newObj.GetLabels(),
+						oldObj.GetLabels(),
+						cmp.Comparer(func(x, y resource.Quantity) bool { return x.Cmp(y) == 0 }),
+					)
+					if diff != "" {
+						logger.Infof("labels has changed for %q. diff=%s", newObj.GetName(), diff)
+						return true
+					}
+					if oldObj.Spec.ObjectBucketName != newObj.Spec.ObjectBucketName {
```

Comment:
AFAIR only Spec can be modified for OBC is the `AdditionalConfig` field others are immutable

Reply:
maybe it should not be done, but when the bucket is actually created this is changing

Reply:
Okay `Spec.ObjectBucketName` is filled for referring the `ob Name`. IMO below case checks for status and if it is `bounded` state then backend bucket will be created. There is no harm in having this check but the below checks gives the required validation

Reply:
when i ran the integration tests, i've seen cases where the state was bounded, but the name was not set yet.
so, to make sure that we won't miss this update, I'm checking on both conditions